plugins/BEdita/Core/src/Job/ServiceRegistry.php

```diff
@@ -49,11 +49,15 @@ public static function get($name)
             return static::$instances[$name];
         }
 
-        $className = Inflector::camelize($name);
-        $fullClassName = App::className($className, 'Job/Service', 'Service');
-        if ($fullClassName === false && strpos($className, '.') === false) {
-            $fullClassName = App::className('BEdita/Core.' . $className, 'Job/Service', 'Service');
+        $plugin = 'BEdita/Core.';
+        if (strpos($name, '.') !== false) {
+            list($plugin, $name) = explode('.', $name);
+            $plugin = Inflector::camelize($plugin) . '.';
         }
+
+        $className = $plugin . Inflector::camelize($name);
+        $fullClassName = App::className($className, 'Job/Service', 'Service');
```

Comment:
I do not quite agree it's the "right" way. Using an approach similar to `cakephp/app` (users actually "fork" the app repository instead of merely cloning it, then install plugins into it) might have benefits as BEdita 4 becomes production ready — versioned `composer.lock`, customized configuration, …
In this case, placing logic in the app would be acceptable.

However, 👍  for now.